header/common.h

```diff
@@ -0,0 +1,235 @@
+
+
+#ifndef WARP_PER_BLOCK
+#define WARP_PER_BLOCK   32
+#endif
+
+#ifndef SIZE_SHARED
+#define SIZE_SHARED 512
+#endif
+
+#ifndef ROWS_PER_THREAD
+#define ROWS_PER_THREAD   1
+#endif
+
+
+// #ifndef GPU_BLK
+// #define GPU_BLK   512
+// #endif
+
+#define PRINT_TIME_ANALYSIS 1 
+
+#define CUDA_EVTS 1
+#define POSIX 2
+
+// #define CLK POSIX
+#define CLK CUDA_EVTS
+
+float clockElapsed(cudaEvent_t evt_start, cudaEvent_t evt_stop);
+
+#if CLK == CUDA_EVTS
+
+	#define CLK_INIT \
+			cudaEvent_t evt_start, evt_stop; \
+			printf("Usando CUDA EVENTS para medir el tiempo\n"); \
+			float t_elap; \
+			cudaEventCreate(&evt_start); \
+			cudaEventCreate(&evt_stop) ;
+
+	#define CLK_START \
+        cudaEventRecord(evt_start, 0);
+
+
+	#define CLK_STOP \
+        cudaEventRecord(evt_stop, 0); 
+
+
+	#define CLK_ELAPSED \
+			t_elap = clockElapsed(evt_start, evt_stop)
+#else
+	#define CLK_INIT \
+			printf("Usando gettimeofday para medir el tiempo\n"); \
+			struct timeval t_i, t_f; \
+			float t_elap
+
+	#define CLK_START \
+			gettimeofday(&t_i,NULL)
+	#define CLK_STOP \
+			gettimeofday(&t_f,NULL) 
+	#define CLK_ELAPSED \
+			t_elap = ((double) t_f.tv_sec * 1000.0 + (double) t_f.tv_usec / 1000.0 - \
+				 	 ((double) t_i.tv_sec * 1000.0 + (double) t_i.tv_usec / 1000.0))
+#endif
+
+
+// #define BENCH_RUN_SOLVE(f,t,pow,s)                                             		 \
+// 	float t;                                                                    	 \
+// 	float pow;                                                                       \
+// 	nvmlAPIRun(s);	                                                                 \
+// 	CLK_START;			                                                             \
+// 	for (int i = 0; i < BENCH_REPEAT; ++i)                                           \
+// 	{	                                                                     	     \
+// 		cudaMemset(d_x, 0, n * sizeof(VALUE_TYPE));                                  \
+// 		f;                                                                           \
+// 	}	                                                                             \
+// 	cudaDeviceSynchronize();                                                         \
+// 	CLK_STOP;       		                                                         \
+// 	pow = nvmlAPIEnd();                                                              \
+// 	t = CLK_ELAPSED/(double)BENCH_REPEAT;                                 			 \
+// 	printf("%20.20s ----- %f ms ----- %f mW", s, t, pow/1000.0 );                    \
+// 	memset(x, 0, n * sizeof(VALUE_TYPE));      	                                 	 \
+// 	cudaMemcpy(x, d_x, n * sizeof(VALUE_TYPE), cudaMemcpyDeviceToHost);              \
+// 	if (validate_x(x, n, s)) { t = -1; }
+
+#define FULL_MASK 0xffffffff
+#define IT_HASH 200
+
+#define BENCH_RUN_SOLVE(f,t,t2,pow,s)                                             					\
+	float t, t2;																					\																				
```

Comment:
what is t, t2?
Can you use more descritpive variable names?
Also cosider declaring them closer to when their being used. (For example before the for statement)